stdlib/websub/src/main/java/org/ballerinalang/net/websub/nativeimpl/InitWebSubSubscriberServiceEndpoint.java

```diff
@@ -110,14 +120,84 @@ public void execute(Context context) {
+                    break;
+                case TOPIC_ID_PAYLOAD_KEY:
+                    populateResourceDetailsByPayload(payloadKeyResourceMap, resourceDetails);
+                    break;
+                default:
+                    populateResourceDetailsByHeaderAndPayload(headerAndPayloadKeyResourceMap, resourceDetails);
+                    if (headerResourceMap != null) {
+                        populateResourceDetailsByHeader(headerResourceMap, resourceDetails);
+                    }
+                    if (payloadKeyResourceMap != null) {
+                        populateResourceDetailsByPayload(payloadKeyResourceMap, resourceDetails);
+                    }
+                    break;
+            }
+        }
+        return resourceDetails;
+    }
+
+    private static void populateResourceDetailsByHeader(BMap<String, BValue> headerResourceMap,
+                                                        HashMap<String, String[]> resourceDetails) {
+        headerResourceMap.getMap().values().forEach(value -> populateResourceDetails(resourceDetails,
+                                                                                     (BRefValueArray) value));
+    }
+
+    private static void populateResourceDetailsByPayload(BMap<String, BMap<String, BValue>> payloadKeyResourceMap,
+                                                         HashMap<String, String[]> resourceDetails) {
+        payloadKeyResourceMap.getMap().values().forEach(mapByKey -> {
+            mapByKey.getMap().values().forEach(value -> populateResourceDetails(resourceDetails,
+                                                                                (BRefValueArray) value));
+        });
+    }
+
+    private static void populateResourceDetailsByHeaderAndPayload(BMap<String, BMap<String, BMap<String, BValue>>>
+                                                                          headerAndPayloadKeyResourceMap,
+                                                                  HashMap<String, String[]> resourceDetails) {
+        headerAndPayloadKeyResourceMap.getMap().values().forEach(mapByHeader -> {
+            mapByHeader.getMap().values().forEach(mapByKey -> {
+                mapByKey.getMap().values().forEach(value -> populateResourceDetails(resourceDetails,
+                                                                                    (BRefValueArray) value));
+            });
+        });
+    }
+
+    private static void populateResourceDetails(HashMap<String, String[]> resourceDetails,
+                                                BRefValueArray resourceDetailTuple) {
+        String resourceName = resourceDetailTuple.getBValue(0).stringValue();
+        BStructureType paramDetails =
+                (BStructureType) ((BTypeDescValue) (resourceDetailTuple).getBValue(1)).value();
+        resourceDetails.put(resourceName,
+                            new String[]{paramDetails.getPackagePath(), paramDetails.getName()});
```

Comment:
Reformat the code. Has unnecessary double lines

Reply:
Fixed with 8da454e, thanks.